### DISCUSSION TITLE
[Junit5] configuration of custom TYPE for @Link

### DISCUSSION THREAD

**Discussion by kat-kan** [allure-framework/allure2#2334] [Questions & Support] [answered]
Hi,

according to the docs, there is possibility to add pattern for custom link type so a full URL is created by adding type and short identifier from value in @Link annotation, unfortunately I cannot make it work. I'd be very grateful for any help.

My setup:
Java, Maven, Junit5, Allure Junit5 v. 2.25.0

Example from the docs:
`@Link(String value="", name="", String url="", String type=CUSTOM_LINK_TYPE)`
Configuration in docs described here: https://allurereport.org/docs/junit5/

What I did: 
- I added allure.link.xyz.pattern = https://example.org/xyz/{} in allure.properties
- I added @Link annotation to test
`@Link(value="XYZ-123", name="XYZ-123", type="xyz")`

Unfortunately, it is not recognized in report and link is not created properly

Another observation is that it works for @Issue annotation 
- allure.link.issue.pattern = https://example.org/xyz/{} in allure.properties
- @Issue(value="XYZ-123")
creates valid, clickable link in report.  So there is a workaround but I'd prefer more flexible option with @Link (possibility to have more than one custom type)

Many thanks in advance!

**Comment by baev** [comment, marked as answer]
I just checked using the latest versions. 

1. https://allurereport.org/start/
2. Select Java -> JUnit 5
3. Keep defaults
4. Add the following test 
    ```java
    package org.example.junit5;
    
    import io.qameta.allure.Link;
    import org.junit.jupiter.api.Test;
    
    class LinkTest {
        @Link(name = "test", type = "xyz")
        @Test
        void customLinks() {
        }
    }
    ```
5. add `allure.link.xyz.pattern=https://xyz.example.org/browse/{}` to `src/test/resources/allure.properties`
6. Run tests (Both IDE or Maven runs should be fine)
7. The link URL is generated successfully
    <img width="461" alt="Screenshot 2024-02-01 at 17 40 16" src="https://github.com/allure-framework/allure2/assets/2149631/f6a9a6b4-d4f0-4ace-b6de-6e26e89c1842">
8. Report looks fine as well
    <img width="1271" alt="Screenshot 2024-02-01 at 17 41 18" src="https://github.com/allure-framework/allure2/assets/2149631/60008310-e6ee-462e-ad09-f2e3c55dfed7">

**Reply by kat-kan** [reply]
Thank you @baev I double checked my dependencies and my Allure Maven reportVersion was 2.24. Turned out that after modifying to 2.25 it works like a charm. Thank you! :)
